metrics/metrics.go

```diff
@@ -0,0 +1,66 @@
+package metrics
+
+import (
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+)
+
+var (
+	MOutgoingMsgs = stats.Int64("pubsub/outgoing_msg_size", "Outgoing message size", "By")
```

Comment:
There are constants for the dimension (last argument), see https://github.com/libp2p/go-libp2p-kad-dht/pull/297/files#diff-e8d467baf1a83863d66232dc09f788e4 for an example.

Also I would recommend removing the prefixing `M`, and not naming your views. See also https://github.com/libp2p/go-libp2p-kad-dht/pull/297/files#diff-e8d467baf1a83863d66232dc09f788e4 for an example. I have some good reasons if you'd like to hear them.

Reply:
@anacrolix 
> I have some good reasons if you'd like to hear them.

Yes.